odoo/tools/set_expression.py

```diff
@@ -137,6 +137,22 @@ def __get_leaf(self, ref: str | int, raise_if_not_found: bool = True) -> Leaf:
             return Leaf(UnknownId(ref), ref)
         return self.__leaves[ref]
 
+    def get_superset_ids(self, ids: list[int]):
+        return {
+            sub_id
+            for id in ids
```

Comment:
General remark: use `id_` instead of `id` (which is a function in standard python).